src/org/opendatakit/briefcase/export/CipherFactory.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.opendatakit.briefcase.export;
+
+import java.io.UnsupportedEncodingException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.apache.commons.codec.binary.Base64;
+import org.opendatakit.briefcase.model.CryptoException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles the initialization of Cipher objects for the decryption of submission
+ * and media files.
+ */
+final class CipherFactory {
+  private static final Logger log = LoggerFactory.getLogger(CipherFactory.class);
+  private static final int IV_BYTE_LENGTH = 16;
+  private final SecretKeySpec symmetricKey;
+  private final byte[] ivSeedArray;
+  private int ivCounter = 0;
+
+  private CipherFactory(String instanceId, byte[] symmetricKeyBytes) throws CryptoException {
+    symmetricKey = new SecretKeySpec(symmetricKeyBytes, "AES/CFB/PKCS5Padding");
+    try {
+      MessageDigest md = MessageDigest.getInstance("MD5");
+      md.update(instanceId.getBytes("UTF-8"));
+      md.update(symmetricKeyBytes);
+      byte[] messageDigest = md.digest();
+      ivSeedArray = new byte[IV_BYTE_LENGTH];
```

Comment:
You may have explained before why you do it this way, but why not initialize in the field declaration?

Reply:
This is just a copy of `org.opendatakit.briefcase.util.CipherFactory`. I honestly don't know why encryption is dealt with this way... I'll rescue some comments from the original, just in case.